packages/amplify-app/src/xcodeHelpers.js

```diff
@@ -15,42 +40,143 @@ async function getXcodeProjectDir() {
   return projDir;
 }
 
-async function addFileToXcodeProj(file, isResource) {
-  const projectPath = await getXcodeProjectDir();
-  // Silently return if not in same directory as xcode project
-  if (!projectPath) {
-    return;
+/**
+ * @private
+ * @param {object} project
+ * @param {string} name
+ * @returns {PBXGroupRef|undefined}
+ */
+function getGroupByName(project, name) {
+  /** @type PBXGroupRef */
+  let rootGroup = null;
+
+  const groups = Object.entries(project.hash.project.objects.PBXGroup);
+  for (let i = 0; i < groups.length; i++) {
+    const [key, value] = groups[i];
+    // console.log(value);
+    if (typeof value !== 'string') {
+      // only the root pbx group can have no name, path or description
+      const isRoot = name === undefined && value.name === undefined && value.path === undefined;
+      const matchesName = typeof name === 'string' && (value.name === name || value.path === name);
+
+      if (isRoot || matchesName) {
+        rootGroup = {
+          uuid: key,
+          pbxGroup: value,
+        };
+        break;
+      }
+    }
   }
-  const myProj = xcode.project(projectPath);
-  return new Promise((resolve, reject) =>
-    myProj.parse(function parseCallback(err) {
-      // hash of the group we add the files to, in this case the root of the xcode project
-      let hash = '';
-      Object.entries(myProj.hash.project.objects.PBXGroup).forEach(entry => {
-        const [key, value] = entry;
-        // only the root pbx group can have no name, path or description
-        if (typeof value !== 'string' && value.name === undefined && value.path === undefined) {
-          hash = key;
-        }
-      });
-
-      if (isResource) {
-        myProj.addPbxGroup([], 'Resources', 'Resources', '"<group>"');
-        myProj.addResourceFile(file, null, hash);
-        myProj.removePbxGroup('Resources');
-      } else {
-        myProj.addFile(file, hash, null);
+  return rootGroup;
+}
+
+/**
+ * @private
+ * @param {object} project
+ * @returns {PBXGroupRef|undefined}
+ */
+function getRootGroup(project) {
+  return getGroupByName(project);
+}
+
+/**
+ * @private
+ * @param {object} project
+ * @param {string} name
+ * @returns {PBXGroupRef}
+ */
+function getOrCreateGroup(project, name) {
+  let group = getGroupByName(project, name);
+  if (!group) {
+    group = project.addPbxGroup([], name, '.');
+    const rootGroup = getRootGroup(project);
+    rootGroup.pbxGroup.children = [{ value: group.uuid, comment: group.pbxGroup.name }, ...rootGroup.pbxGroup.children];
+  }
+  return group;
+}
+
+/**
+ * @private
+ * @param {PBXGroupRef} group
+ * @param {string} name
+ * @returns {boolean}
+ */
+function groupHasFile(group, name) {
+  return group.pbxGroup.children.filter(child => child.comment === name).length > 0;
+}
+
+/**
+ * @public
+ * @return {Promise<void>}
+ */
+async function addDataStoreFiles() {
```

Comment:
We seem to be adding more than just dataStoreFiles in this function (but the function name is addDataStoreFiles).... maybe we should rename this function to something like addAmplifyFiles() which calls smaller functions like addGeneratedModels, addConfigGroup, etc..

Reply:
amplify-app is not resilient enough to work with things outside of `Amplify.DataStore`. I can rename the function, but I'm reluctant to break it down into smaller functions until we actually need because of Promises/Error Handlings/etc.

My goal in mentioning DataStore here is find-ability, `amplify-app` is primarily used for DataStore but there's no mention of it.

Reply:
updated to `addAmplifyFiles()` but kept the implementation, let me know what you think